test/helpers.js

```diff
@@ -283,6 +306,36 @@ var compareItem = function(actual, expected, message, options, properties) {
     }
 };
 
+/**
+ * Run each callback in a separated canvas context and compare both outputs.
+ * This can be used to do selection drawing tests as it is not possible with
+ * comparePixels() method which relies on the item.rasterize() method which
+ * ignores selection.
+ * @param width the width of the canvas
+ * @param height the height of the canvas
+ * @param callback1 the function producing the expected result
+ * @param callback2 the function producing the actual result
+ */
+var compareCanvas = function(width, height, callback1, callback2) {
+    function getImageData(width, height, callback) {
+        var canvas = CanvasProvider.getCanvas(width, height);
+        var project = new Project(canvas);
+        callback();
+        project.view.update();
+        var imageData = canvas.getContext('2d').getImageData(0, 0, width, height);
+        CanvasProvider.release(canvas);
+        project.remove();
+        return imageData;
+    }
+
+    compareImageDatas(
```

Comment:
@sasensi the plural of `data` is ...  `data` :) Let's call it `compareImageData()`. It's clear that you need to provide two since you compare them.

Reply:
Ok :D